featurebyte/worker/task/catalog_online_store_update.py

```diff
@@ -0,0 +1,98 @@
+        offline_store_feature_table_service: OfflineStoreFeatureTableService,
+        feast_registry_service: FeastRegistryService,
+        feast_feature_store_service: FeastFeatureStoreService,
+        catalog_service: CatalogService,
+    ):
+        super().__init__()
+        self.feature_materialize_service = feature_materialize_service
+        self.offline_store_feature_table_service = offline_store_feature_table_service
+        self.feast_registry_service = feast_registry_service
+        self.feast_feature_store_service = feast_feature_store_service
+        self.catalog_service = catalog_service
+
+    async def get_task_description(self, payload: CatalogOnlineStoreInitializeTaskPayload) -> str:
+        if payload.online_store_id is not None:
+            return f'Initialize online store "{payload.online_store_id}" for catalog {payload.catalog_id}'
+        return f"Disabling online store for catalog {payload.online_store_id}"
+
+    async def execute(self, payload: CatalogOnlineStoreInitializeTaskPayload) -> Any:
+        logger.info(f"Starting task: {self.get_task_description(payload)}")
+
+        if payload.online_store_id is not None:
+            await self._run_materialize(payload)
+
+        logger.info(
+            "Updating online store for catalog",
+            extra={"online_store_id": payload.online_store_id, "catalog_id": payload.catalog_id},
+        )
+        await self.catalog_service.update_online_store(
+            document_id=payload.catalog_id,
+            data=CatalogOnlineStoreUpdate(online_store_id=payload.online_store_id),
+        )
+
+    async def _run_materialize(self, payload: CatalogOnlineStoreInitializeTaskPayload) -> None:
+        feast_registry = await self.feast_registry_service.get_feast_registry_for_catalog()
+        if feast_registry is None:
+            return None
+        feast_feature_store = await self.feast_feature_store_service.get_feast_feature_store(
+            feast_registry_id=feast_registry.id,
+            online_store_id=payload.online_store_id,
+        )
+        session = None
+        async for feature_table_model in self.offline_store_feature_table_service.list_documents_iterator(
+            {}
+        ):
+            logger.info(
+                f"Updating online store for offline feature store table {feature_table_model.name}",
+                extra={
+                    "online_store_id": payload.online_store_id,
+                    "catalog_id": payload.catalog_id,
+                },
+            )
+            if session is None:
+                session = await self.feature_materialize_service._get_session(  # pylint: disable=protected-access
+                    feature_table_model
+                )
+            await self.feature_materialize_service.update_online_store(
+                feature_store=feast_feature_store,
+                feature_table_model=feature_table_model,
+                session=session,
+            )
```

Comment:
If the task is expect to run for some time, would be good to call `self.task_progress_updater.update_progress` to update progress.